mkosi/__init__.py

```diff
@@ -941,13 +941,14 @@ def install_systemd_boot(context: Context) -> None:
         return
 
     directory = context.root / "usr/lib/systemd/boot/efi"
-    if not directory.exists() or not any(directory.iterdir()):
+    signed = context.config.shim_bootloader == ShimBootloader.signed
+    if not directory.glob("*.efi.signed" if signed else "*.efi"):
         if context.config.bootable == ConfigFeature.enabled:
-            die("A EFI bootable image with systemd-boot was requested but systemd-boot was not found at "
-                f"{directory.relative_to(context.root)}")
+            die(f"A EFI bootable image with systemd-boot was requested but a {'signed' if signed else ''} "
```

Comment:
Nit: 
```suggestion
            die(f"A EFI bootable image with systemd-boot was requested but a {'signed ' if signed else ''}"
```
otherwise there is a double space after "a" in the unsigned case.